src/test/regress/expected/failure_distributed_results.out

```diff
@@ -0,0 +1,132 @@
+ t
+(1 row)
+
+SELECT count(*), sum(x) FROM
+  read_intermediate_results('{test_from_100802_to_1,test_from_100802_to_2}'::text[],'binary') AS res (x int);
+ count | sum
+---------------------------------------------------------------------
+    15 | 863
+(1 row)
+
+ROLLBACk;
+-- with failure, results from 100802 should be retried and succeed on 57637
+SELECT citus.mitmproxy('conn.onQuery(query="worker_partition_query_result.*test_from_100802").kill()');
+ mitmproxy
+---------------------------------------------------------------------
+
+(1 row)
+
+BEGIN;
+CREATE TABLE distributed_result_info AS
+  SELECT resultId, nodeport, rowcount, targetShardId, targetShardIndex
+  FROM partition_task_list_results('test', $$ SELECT * FROM source_table $$, 'target_table')
+          NATURAL JOIN pg_dist_node;
+WARNING:  connection error: localhost:xxxxx
+DETAIL:  server closed the connection unexpectedly
+	This probably means the server terminated abnormally
+	before or while processing the request.
+SELECT * FROM distributed_result_info ORDER BY resultId;
+       resultid        | nodeport | rowcount | targetshardid | targetshardindex
+---------------------------------------------------------------------
+ test_from_100800_to_0 |     9060 |       22 |        100805 |                0
+ test_from_100801_to_0 |    57637 |        2 |        100805 |                0
+ test_from_100801_to_1 |    57637 |       15 |        100806 |                1
+ test_from_100802_to_1 |    57637 |       10 |        100806 |                1
+ test_from_100802_to_2 |    57637 |        5 |        100807 |                2
+ test_from_100803_to_2 |    57637 |       18 |        100807 |                2
+ test_from_100803_to_3 |    57637 |        4 |        100808 |                3
+ test_from_100804_to_3 |     9060 |       24 |        100808 |                3
+(8 rows)
+
+-- fetch from worker 2 should fail
+SAVEPOINT s1;
+SELECT fetch_intermediate_results('{test_from_100802_to_1,test_from_100802_to_2}'::text[], 'localhost', :worker_2_port) > 0 AS fetched;
+ERROR:  could not open file "base/pgsql_job_cache/xx_x_xxx/test_from_100802_to_1.data": No such file or directory
+CONTEXT:  while executing command on localhost:xxxxx
+ROLLBACK TO SAVEPOINT s1;
+-- fetch from worker 1 should succeed
+SELECT fetch_intermediate_results('{test_from_100802_to_1,test_from_100802_to_2}'::text[], 'localhost', :worker_1_port) > 0 AS fetched;
+ fetched
+---------------------------------------------------------------------
+ t
+(1 row)
+
+-- make sure the results read are same as the previous transaction block
+SELECT count(*), sum(x) FROM
+  read_intermediate_results('{test_from_100802_to_1,test_from_100802_to_2}'::text[],'binary') AS res (x int);
+ count | sum
+---------------------------------------------------------------------
+    15 | 863
+(1 row)
+
+ROLLBACk;
+SET search_path TO 'public';
+RESET citus.shard_replication_factor;
+RESET citus.shard_count;
+RESET citus.task_assignment_policy;
```

Comment:
Note that these are generally unnecessary since the session ends at the end of the file.